File: AtCoder/Browns/cpp/ABC030-B-Tokeiban.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N, M;

int main() {
  cin >> N >> M;
  double S = N * 2 * M_PI / 12 + M * 2 * M_PI / 60 / 12;
  double L = M * 2 * M_PI / 60;
  double c = cos(S) * cos(L) + sin(S) * sin(L);
  double d = acos(c) * 360 / (2 * M_PI);
  if (d < 0) d += 360;
  printf("%.5f\n", min(d, 360 - d));
}

File: AtCoder/Browns/python/ABC030-B-Tokeiban.py
from math import pi, cos, sin, acos, degrees
N, M = map(int, input().split())
S = N * 2 * pi / 12 + M * 2 * pi / 60 / 12
L = M * 2 * pi / 60
rad = cos(S) * cos(L) + sin(S) * sin(L)
deg = degrees(acos(rad))
if deg < 0:
    deg += 360
print(f"{min(deg, 360 - deg):.5f}")
